core/sql/optimizer/NormRelExpr.cpp

```diff
@@ -6995,11 +6995,6 @@ NABoolean RelRoot::isUpdatableBasic(NABoolean isView,
   // QSTUFF
 
     {
-      // if child is a FirstN node, skip it.
-      if ((child(0)->castToRelExpr()->getOperatorType() == REL_FIRST_N) &&
-	  (child(0)->child(0)))
-	scan = (Scan *)child(0)->child(0)->castToRelExpr();
-      else
 	scan = (Scan *)child(0)->castToRelExpr();
```

Comment:
There may be a few isolated cases where the FirstN node is added  during preCodeGen. Please see GenPreCode.cpp RelRoot::preCodeGen(). The example given there about Order by where sort is added in optimizer, or a FirstN where the N value is to be specified with a param seem to be cases where we would add the FirstN later. Will current change cause such views to marked as updateable? This is minor and could be resolved later, if any change is necessary.

Reply:
I will look into it.

Reply:
Thanks, Suresh, for the pointer. While trying to go through the RelRoot::preCodeGen() logic and cause a FirstN node to be inserted, I ran into this bug:

1. I created a table t1 with six rows.
2. I did "create view v1 as select [first 5] * from t1 order by a;"  (I wanted to see if the order by clause was permitted and if so would it circumvent the updatable view check. It did!)
3. I did "update v1 set b = 6". It updated 6 rows.

So, it appears that the updatable view test is not as complete as it needs to be.

I have written a new JIRA, https://issues.apache.org/jira/browse/TRAFODION-2840, to track this problem.